### DISCUSSION TITLE
*elastalert missing

### DISCUSSION THREAD

**Discussion by iqworks** [Security-Onion-Solutions/securityonion#8961] [Unsupported Versions] [answered]
Hi, how do I install or re-install when I see elastalert missing while checking container statuses?
  thanks for any help or advice

**Comment by dougburks** [comment, marked as answer]
This usually means that there is a problem with the ElastAlert configuration. Have you changed the ElastAlert or Playbook configuration in any way?

You can check diagnostic logs for additional clues as shown at https://docs.securityonion.net/en/2.3/elastalert.html#diagnostic-logging.

If all else fails, you can perform a fresh installation as we discussed in a separate thread.

**Comment by iqworks** [comment]
Thanks for getting back,  I didn’t I will review this link before I rebuild. By the way, dell took over my laptop and cleared my cache and temporary files. I let SO run over the weekend and today, Monday it is still running and not using as much CPU. I will add that to the other post as well. 

Thanks as always

**Reply by iqworks** [reply]
Hi Doug, I am still using “securityonion-2.3.160-20220829.iso”. I was following this link "
 https://docs.securityonion.net/en/2.3/elastalert.html#diagnostic-logging ".
 I can see these logs. But no elastalert.log.
![image](https://user-images.githubusercontent.com/5569263/198348544-df05c1a1-248d-4d25-aacc-23fd2e2f9c88.png)

but I use the “vi” command on stderr.log and I don’t see anything? Might this be because elastalert container is MISSING? 
![image](https://user-images.githubusercontent.com/5569263/198348646-df024131-1c02-4335-8cc3-cc6938529732.png)
I also tried these two as well. The VM that works displays info for docker.  
![image](https://user-images.githubusercontent.com/5569263/198348767-bdcb92b8-7286-4344-b734-eb7c4310d76f.png)

![image](https://user-images.githubusercontent.com/5569263/198348825-e2c6b2be-f4dc-49c2-b188-4eea2c478be0.png) 

I must be looking in the wrong places? I think it is what it says “MISSING”. I cannot seem to find how it could be re-installed? Or, do I have to create a new VM period?

Also I never did anything to the playbook or elastalert config's. 

However, this error usually occurs BEFORE the next time I power on and then I see the MISSING status. 
![image](https://user-images.githubusercontent.com/5569263/198350366-f1becf06-2f31-4683-900e-fd473dac7f51.png)
VMware says : 
https://kb.vmware.com/s/article/67623 
So I wonder if this might cause the missing elastalert MISSING? I am investigating this as well. 

thanks for your time and advice

**Reply by dougburks** [reply]
> I can see these logs. But no elastalert.log.

Hmm...looks like it's at the bottom of the first column:
![elastalert log](https://user-images.githubusercontent.com/1659467/198572251-2df42870-f780-43da-beac-ae5eb49eccfa.png)

You can also check `sudo docker images |grep elastalert` to see if the Docker image exists.

If all else fails, you can try a fresh installation of the latest version:
https://blog.securityonion.net/2022/10/security-onion-23181-now-available.html

**Reply by iqworks** [reply]
Yep, I meant to say it was empty. 
![image](https://user-images.githubusercontent.com/5569263/198614965-faddbe76-3587-4388-89ec-9641a9004210.png)
Here is the elastalert.log from the VM that works 
![image](https://user-images.githubusercontent.com/5569263/198641624-e068a1ab-5229-4d98-a94e-bca0b8cf4171.png)
So I guess its TRULY gone! 
 I will try the lastest version, but the problem is that it is happening on version 150,160 and 170.  HA, its only a program I will continue my research. It might be something that chokes in VMware, but I dont know how just elastalert can be deleted from that, its just a suspicion. 
  I will keep this post informed.   
Thanks again Doug.

**Reply by dougburks** [reply]
> So I guess its TRULY gone!

I doubt that it's truly gone. Did you try the command I included in my previous reply?
```
sudo docker images |grep elastalert
```

> I will try the lastest version, but the problem is that it is happening on version 150,160 and 170. HA, its only a program I will continue my research. It might be something that chokes in VMware, but I dont know how just elastalert can be deleted from that, its just a suspicion.

I don't think it's a VMware issue. We test all releases in VMware and ElastAlert works fine for us:
```
Checking Docker status

    Docker ---------------------------------------------------------- [ OK ]    

Checking container statuses

    so-aptcacherng -------------------------------------------------- [ OK ]    
    so-curator ------------------------------------------------------ [ OK ]    
    so-dockerregistry ----------------------------------------------- [ OK ]    
    so-elastalert --------------------------------------------------- [ OK ]    
    so-elasticsearch ------------------------------------------------ [ OK ]    
    so-filebeat ----------------------------------------------------- [ OK ]    
    so-grafana ------------------------------------------------------ [ OK ]    
    so-idstools ----------------------------------------------------- [ OK ]    
    so-influxdb ----------------------------------------------------- [ OK ]    
    so-kibana ------------------------------------------------------- [ OK ]    
    so-kratos ------------------------------------------------------- [ OK ]    
    so-logstash ----------------------------------------------------- [ OK ]    
    so-nginx -------------------------------------------------------- [ OK ]    
    so-redis -------------------------------------------------------- [ OK ]    
    so-sensoroni ---------------------------------------------------- [ OK ]    
    so-soc ---------------------------------------------------------- [ OK ]    
    so-soctopus ----------------------------------------------------- [ OK ]    
    so-steno -------------------------------------------------------- [ OK ]    
    so-strelka-backend ---------------------------------------------- [ OK ]    
    so-strelka-coordinator ------------------------------------------ [ OK ]    
    so-strelka-filestream ------------------------------------------- [ OK ]    
    so-strelka-frontend --------------------------------------------- [ OK ]    
    so-strelka-gatekeeper ------------------------------------------- [ OK ]    
    so-strelka-manager ---------------------------------------------- [ OK ]    
    so-suricata ----------------------------------------------------- [ OK ]    
    so-telegraf ----------------------------------------------------- [ OK ]    
    so-zeek --------------------------------------------------------- [ OK ]    
```

**Reply by iqworks** [reply]
Hi Doug, it looks like it has a connection issue with connecting to my SOC IP address. The connection gets refused to so-elastalert I think? I guess it goes to this IP address to get stuff for elastalert?
yep that “sudo docker images |grep elastalert”.   
![image](https://user-images.githubusercontent.com/5569263/199363187-b5435491-0b2a-4474-aa60-cfc1e9130b62.png)

I also saw “sudo docker logs so-elastalert” which showed me that there 
Appears to be an issue with connecting?
 
![image](https://user-images.githubusercontent.com/5569263/199363369-191442ce-20bf-4328-b26d-55214bff2d5b.png) 
![image](https://user-images.githubusercontent.com/5569263/199363394-2bf01321-b025-4603-803f-5d2fea7d4a92.png)
![image](https://user-images.githubusercontent.com/5569263/199363417-2acff301-fd42-4140-976f-aaabd47ed800.png)
 
Run sudo docker ps -a again and i can see so-elastalert has exited.
![image](https://user-images.githubusercontent.com/5569263/199363544-c090a75c-fb53-4ee4-a144-0cf85b08e5d4.png)

 
I ran sudo docker logs so-elastalert again and I saw there were issues with connecting? 
![image](https://user-images.githubusercontent.com/5569263/199363567-4490a19b-4995-49b8-885b-3ded3e547441.png) 
![image](https://user-images.githubusercontent.com/5569263/199363634-ed901b17-cf73-4729-8ee3-f76aadce4637.png)
![image](https://user-images.githubusercontent.com/5569263/199363668-7703f5ce-4e0f-4733-b34d-99c815b3338d.png)
![image](https://user-images.githubusercontent.com/5569263/199363683-bbb6dd49-6e83-45ef-8fa2-59450dfbb2a6.png)
![image](https://user-images.githubusercontent.com/5569263/199363728-f8dba825-45ab-4755-ade5-73280a79b67b.png)



I ran sudo docker logs so-elastalert again and saw this 
![image](https://user-images.githubusercontent.com/5569263/199363756-f8d482c2-e996-488b-9264-6efc0b9dddd9.png)

I saw  that elastalert and soctopus containers wind up starting at the same time.
At one point, so-elastalert shows OK, and so-soctopus says STARTING. 
But then it says so-elastalert changes from OK to STARTING. Then it goes MISSING.
Then this was very interesting.

 
I have to do a little work on another project for a few days, but I will get back to this and look at the connection issue. But I believe your right and this might be a bad VM. I am creating a 2.3.181-20221021 VM.
Thanks as always

**Reply by dougburks** [reply]
> Hi Doug, it looks like it has a connection issue with connecting to my SOC IP address. The connection gets refused to so-elastalert I think? I guess it goes to this IP address to get stuff for elastalert?

ElastAlert connects to Elasticsearch and runs queries. To do so, it connects to the main IP address of your Security Onion installation.

What is the current IP address of your Security Onion VM? You can check with the `ip a` command.

Is it `192.168.16.128` or has it changed since you ran Setup?

Did you configure for static IP address or DHCP?

> I have to do a little work on another project for a few days, but I will get back to this and look at the connection issue. But I believe your right and this might be a bad VM. I am creating a 2.3.181-20221021 VM.

Please make sure you configure a static IP address and avoid changing that IP address after installation.

**Reply by iqworks** [reply]
Thanks Doug, the thing is that my IP address has stayed the same for months?  But if elastalert needs to feed elastasearch on my SOC and the communication is refused, its like it just lost its connection, or cannot establish it? 
  I have reserved the IP addresses from the current IP on each of my laptops. So now each one has its current IP reserved. 
  I guess somewhere along the way to get all docker containers to turn statuses OK, the SOC needs to use my laptops static IP to get a different type of connection between the SOC and my laptop?  I wonder where the SOC needs to use my laptops IP address instead of the IP address that gets refused when elastalert tries to connect?
Thanks again for your help

**Reply by dougburks** [reply]
This has nothing to do with the IP address of your laptop. Your Security Onion VM should have its own static IP address separate from the IP address of your laptop. ElastAlert (inside the Security Onion VM) should be connecting to Elasticsearch (inside the Security Onion VM) using the static IP address of the Security Onion VM.

**Reply by iqworks** [reply]
Hi Doug, thanks as always for your time. Yesterday I reserved my current IP address so it will never change, so I can use this as my static IP when I create  my next VM. 
  So, by “(inside the Security Onion VM” you mean the SOC or my VM?
  
  But I was referring to this message :
![image](https://user-images.githubusercontent.com/5569263/200028231-aeaa3a7a-ee5c-4ea2-941b-1c5213593149.png)

  The IP address here in this error message is my SOC? And it says “during handling of the above exception, another exception has occurred”  and it mentions elastalert, It looks like elastalert wants to connect to my SOC?
Thanks again

**Reply by dougburks** [reply]
As I mentioned in my previous reply, your Security Onion VM should have its own static IP address and ElastAlert (inside the Security Onion VM) should be connecting to Elasticsearch (inside the Security Onion VM) using that IP address. That IP address allows Security Onion to connect to the outside world for things like updates and it allows you to connect from your web browser to the Security Onion Console. Security Onion Console is just a web server running inside the Security Onion VM.

**Comment by iqworks** [comment]
Thanks Doug for the explanation. I see now that when I call up the SOC, it is actually a server running on my created VM. 

  I also see that I need a unique reserved IP address different from the one assigned by my cable internet provider. 

  I also see that this static IP address is used when elastalert needs to send data to elastasearch that is also in my created VM. 

  I am going to set this up. 

Thanks as always for your time.
